### PR TITLE
Add hooks

### PR DESCRIPTION
**Pull request by rcqx**
## Comments
* Refactor Calculator.js component from class to function
* Make use of hooks to create and update state within the function by importing useEffect 
* Create a button component that accepts props (className, text, onClick)
* Defines default props values and propTypes by installing and importing "PropTypes" in the button component

image

### CODE REVIEW

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,64 +1,55 @@
-import React from 'react';
+import { useState } from 'react';
 import calculate from '../logic/calculate';
+import Button from './Button';
 
 // eslint-disable-next-line react/prefer-stateless-function
-class Calculator extends React.Component {
```
Comment (Wes-Isaac) [inline src/components/Calculator.js]: - [ ] Great job so far, it is better to solve the error than disable the linter, kindly fix the errors, if any.

File: src/components/Calculator.js [hunk 2]
```diff
@@ -1,64 +1,55 @@
...
-          <div role="button" className="grid-item operator" onKeyDown={this.keydown} tabIndex={0} onClick={this.compute}>+</div>
-          <div role="button" className="grid-zero" onKeyDown={this.keydown} tabIndex={0} onClick={this.compute}>0</div>
-          <div role="button" className="grid-item" onKeyDown={this.keydown} tabIndex={0} onClick={this.compute}>.</div>
-          <div role="button" className="grid-item operator" onKeyDown={this.keydown} tabIndex={0} onClick={this.compute}>=</div>
+            {values.next}
+          </span>
         </div>
+        <Button className="grid-item" text="AC" onClick={compute} />
+        <Button className="grid-item" text="+/-" onClick={compute} />
+        <Button className="grid-item" text="%" onClick={calculate} />
```
Comment (Wes-Isaac) [inline src/components/Calculator.js]: - [ ] Great job, there is no `calculate` method in this component, maybe it's a typo kindly fix it. you got this 💪.

**Review by Wes-Isaac — CHANGES_REQUESTED** [review]
Hi @jr-cast, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Linters are set up properly ✔️.
- Great summary😲. 
- Project works as expected 😮

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by hiromataba — APPROVED** [review]
# Status: Approved :tada: :tada: :tada: 
Hi @jr-cast     :wave: :wave: 

200w  

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**